File: CS/NonPersistentObjectsDemo.Module/NonPersistentObjectFactory.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DevExpress.Data.Filtering;

namespace NonPersistentObjectsDemo.Module {

    public abstract class NonPersistentObjectFactoryBase {
        public abstract object GetObjectByKey(Type objectType, object key);
        public abstract IEnumerable GetObjectKeys(Type objectType, CriteriaOperator criteria, IList<DevExpress.Xpo.SortProperty> sorting);
        public virtual void SaveObjects(ICollection toInsert, ICollection toUpdate, ICollection toDelete) { }
    }
}

File: VB/NonPersistentObjectsDemo.Module/NonPersistentObjectFactory.vb
Imports System
Imports System.Collections
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports DevExpress.Data.Filtering

Namespace NonPersistentObjectsDemo.Module

	Public MustInherit Class NonPersistentObjectFactoryBase
		Public MustOverride Function GetObjectByKey(ByVal objectType As Type, ByVal key As Object) As Object
		Public MustOverride Function GetObjects(ByVal objectType As Type, ByVal criteria As CriteriaOperator, ByVal sorting As IList(Of DevExpress.Xpo.SortProperty)) As IEnumerable
		Public Overridable Sub SaveObjects(ByVal toInsert As ICollection, ByVal toUpdate As ICollection, ByVal toDelete As ICollection)
		End Sub
	End Class
End Namespace
